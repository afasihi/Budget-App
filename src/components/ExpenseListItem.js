import React from 'react';
import { NavLink } from 'react-router-dom'
import moment from 'moment';
import numeral from 'numeral'

const ExpenseListItem = ({ description, amount, createdAt, id }) => (
	<div>
		<NavLink to={`/edit/${id}`}>
			<h3>{description}</h3>
		</NavLink>
		<p>
			{numeral(amount / 100).format('$0,0.00')} 
			- 
			{moment(createdAt).format('MMMM Do, YYYY')}</p>

	</div>
)

export default ExpenseListItem