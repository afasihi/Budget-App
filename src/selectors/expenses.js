import moment from 'moment'

export default (expenses, { sortBy, endDate, startDate, text }) => {
	return expenses.filter((expense) => {
		const createdAtMoment = moment(expense.createdAt)
		const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true
		const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true
		const testMatch = expense.description.toLowerCase().includes(text.toLowerCase())

		return startDateMatch && endDateMatch && testMatch
	}).sort((a, b) => {
		if (sortBy === 'date')
			return a.createdAt < b.createdAt ? 1 : -1
		else if (sortBy === 'amount')
			return a.amount < b.amount ? 1 : -1
	})
}