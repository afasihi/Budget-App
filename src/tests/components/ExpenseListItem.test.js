import React from 'react'
import { shallow } from 'enzyme'
import ExpenseListItem from '../../components/ExpenseListItem'
import expenses from '../fixtures/expenses'

test('should render ExoenseList with expenses', () => {
	const wrapper = shallow(<ExpenseListItem {...expenses[0]} />)
	expect(wrapper).toMatchSnapshot()
})