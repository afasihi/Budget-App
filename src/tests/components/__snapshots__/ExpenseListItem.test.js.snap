// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExoenseList with expenses 1`] = `
<div>
  <NavLink
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </NavLink>
  <p>
    $1.95
    -
    January 1st, 1970
  </p>
</div>
`;
